### PR TITLE
NDRS-833: Unit test handling incoming finality signatures.

### PR DESCRIPTION
**Pull request by goral09**
First PR, out of n, that adds tests to the `LinearChain` high-level API.

### CODE REVIEW

File: node/src/components/linear_chain/state.rs [hunk 1]
```diff
@@ -367,3 +368,141 @@ impl LinearChain {
...
+
+    #[test]
+    fn pending_sig_rejected() {
+        let mut rng = TestRng::new();
+        let protocol_version = ProtocolVersion::V1_0_0;
+        let mut lc = LinearChain::new(protocol_version, 1u64, 1u64);
+        let block_hash = BlockHash::random(&mut rng);
+        let valid_sig = FinalitySignature::random_for_block(block_hash, 0);
+        let handle_sig_outcomes = lc.handle_finality_signature(Box::new(valid_sig.clone()), false);
+        assert_eq!(handle_sig_outcomes.len(), 1);
```
Comment (afck) [inline node/src/components/linear_chain/state.rs:420]: Why not check that it's the expected outcome? (`LoadSignatures`?)

**Review by afck — APPROVED** [review]

File: node/src/components/linear_chain/state.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/linear_chain/state.rs:420]: I think I couldn't get the compiler to work as I wanted but then learned how to do it later. I will change that.

File: node/src/components/linear_chain/state.rs [hunk 2]
```diff
@@ -367,3 +368,141 @@ impl LinearChain {
...
+        let mut rng = TestRng::new();
+        let protocol_version = ProtocolVersion::V1_0_0;
+        let mut lc = LinearChain::new(protocol_version, 1u64, 1u64);
+        let block = Block::random(&mut rng);
+        let execution_results = HashMap::new();
+        let new_block_outcomes =
+            lc.handle_new_block(Box::new(block.clone()), execution_results.clone());
+        let block_hash = *block.hash();
+        match &*new_block_outcomes {
+            [Outcome::StoreBlock(outcome_block), Outcome::StoreExecutionResults(outcome_block_hash, outcome_execution_results)] =>
```
[the remark this answers is not in the document]
Comment (goral09) [reply node/src/components/linear_chain/state.rs:390]: If I change it then `cargo fmt` rearranges it back to this.

File: node/src/components/linear_chain/state.rs [hunk 3]
```diff
@@ -25,6 +25,7 @@ pub(crate) struct LinearChain {
     auction_delay: u64,
     unbonding_delay: u64,
 }
+#[derive(Debug)]
```
[the remark this answers is not in the document]
Comment (goral09) [reply node/src/components/linear_chain/state.rs:29]: Done in [`6f5f46c` (#1284)](https://github.com/CasperLabs/casper-node/pull/1284/commits/6f5f46c35c509dfb3f246460187696ebb353bc9b)

File: node/src/components/linear_chain/state.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/linear_chain/state.rs]: [`6f5f46c` (#1284)](https://github.com/CasperLabs/casper-node/pull/1284/commits/6f5f46c35c509dfb3f246460187696ebb353bc9b)

File: node/src/components/linear_chain/state.rs [hunk 4]
```diff
@@ -367,3 +368,141 @@ impl LinearChain {
...
+        let execution_results = HashMap::new();
+        let new_block_outcomes =
+            lc.handle_new_block(Box::new(block.clone()), execution_results.clone());
+        let block_hash = *block.hash();
+        match &*new_block_outcomes {
+            [Outcome::StoreBlock(outcome_block), Outcome::StoreExecutionResults(outcome_block_hash, outcome_execution_results)] =>
+            {
+                assert_eq!(&**outcome_block, &block);
+                assert_eq!(outcome_block_hash, &block_hash);
+                assert_eq!(outcome_execution_results, &execution_results);
```
[the remark this answers is not in the document]
Comment (goral09) [reply node/src/components/linear_chain/state.rs:395]: Very cumbersome. I will see if I can add it in a later PRs and then I'll change this.

**Comment by goral09** [comment]
bors merge

File: node/src/components/linear_chain/state.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (afck) [reply node/src/components/linear_chain/state.rs:395]: It's probably not actually important; was just a thought.

**Pull request merged**